test_tf.py

```diff
@@ -0,0 +1,153 @@
+    def test_get_sub_command_case12(self):
+        required = {"INPUT_TEMPLATE-PATH",  "INPUT_REGION"}
+        assertRaises(ValueError, get_sub_command, required, True)
+
+    def test_get_sub_command_case13(self):
+        tft = os.getcwd() + './tf_validate.json'
+        os.environ['INPUT_TEMPLATE-PATH'] = tft
+        os.environ['INPUT_REGION'] = 'us-west-2'
+        required = {"INPUT_TEMPLATE-PATH",  "INPUT_REGION"}
+        expected = ['--template-path', tft, '--region', 'us-west-2']
+        flags = get_sub_command(required, True)
+        assert set(flags) == set(expected)
+
+    # test_get_treat_findings_as_non_blocking_flag_case1: failure expected because os.environ[TREAT_FINDINGS_AS_NON_BLOCKING] is not set
+    @unittest.expectedFailure
+    def test_get_treat_findings_as_non_blocking_flag_case14(self):
+        policy_check = CHECK_NO_NEW_ACCESS
+        assertRaises(ValueError, get_treat_findings_as_non_blocking_flag, policy_check)
+        
+    def test_get_treat_findings_as_non_blocking_flag_case15(self):
+        policy_check = VALIDATE_POLICY
+        result = get_treat_findings_as_non_blocking_flag(policy_check)
+        assert result == "" 
+        
+    def test_get_treat_findings_as_non_blocking_flag_case16(self):
+        policy_check = CHECK_ACCESS_NOT_GRANTED
+        os.environ[TREAT_FINDINGS_AS_NON_BLOCKING] = 'True'
+        result = get_treat_findings_as_non_blocking_flag(policy_check)
+        assert result == ['--treat-findings-as-non-blocking']
+
+    def test_get_treat_findings_as_non_blocking_flag_case17(self):
+        policy_check = CHECK_ACCESS_NOT_GRANTED
+        os.environ[TREAT_FINDINGS_AS_NON_BLOCKING] = 'False'
+        result = get_treat_findings_as_non_blocking_flag(policy_check)
+        assert result == "" 
+
+    def test_build_command_case18(self):
+        policy_check = VALIDATE_POLICY
+        tft = os.getcwd() + './tf_validate.json'
+        os.environ['INPUT_TEMPLATE-PATH'] = tft 
+        os.environ['INPUT_REGION'] = 'us-west-2'
+        os.environ['TREAT_FINDINGS_AS_NON_BLOCKING'] = 'False'
+        os.environ['INPUT_IGNORE-FINDING'] = 'PASS_ROLE_WITH_STAR_IN_RESOURCE'
+        required = {"INPUT_TEMPLATE-PATH",  "INPUT_REGION"}
+        optional = {"INPUT_IGNORE-FINDING"}
+        command = build_command(policy_check, required, optional)
+        # print("command {}".format(command))
+        expected = set([CLI_POLICY_VALIDATOR, 'validate', '--template-path', tft, '--region', 'us-west-2', '--ignore-finding', 'PASS_ROLE_WITH_STAR_IN_RESOURCE'])
+        command_set = set(command)
+        assert command_set == expected
+
+#     def test_execute_command_case19(self):
+#         tft = os.getcwd() + '/tf_validate.json'
+#         cfg = os.getcwd() + '/default.yaml'
+#         command = [CLI_POLICY_VALIDATOR, 'validate', '--config', cfg, '--template-path', tft, '--region', 'us-west-2', '--treat-finding-type-as-blocking', 'ERROR']
+#         result = execute_command(command)
```

Comment:
this is also unit testable, but we'd need to mock the `subprocess.run` method to not make it actually run the command